llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -30624,6 +30650,10 @@ bool AArch64TargetLowering::isTargetCanonicalConstantNode(SDValue Op) const {
   return Op.getOpcode() == AArch64ISD::DUP ||
          Op.getOpcode() == AArch64ISD::MOVI ||
          Op.getOpcode() == AArch64ISD::MOVIshift ||
```

Comment:
Hi David and Simon,

Following the other thread about adding fneg(zero) as a canonical constant pattern, I am still a bit confused about what I should do and expect here. 

1) Shall I add something along the lines of the following:
```suggestion
         Op.getOpcode() == AArch64ISD::MOVIshift ||
         (Op.getOpcode() == ISD::FNEG &&
         Op.getOperand(0).getOpcode() == AArch64ISD::MOVI &&
         Op.getOperand(0).getConstantOperandVal(0) == 0) ||
```

Please see my other question my other comment in file `llvm/test/CodeGen/AArch64/urem-vector-lkk.ll`.

Reply:
That's an excellent question that I don't know the answer to! I did not see one in the code, so I didn't add it. According to David, both are type f64, so we are fine...